crates/shared/src/price_estimation/native_price_cache.rs

```diff
@@ -24,12 +24,18 @@ struct Metrics {
     native_price_cache_background_updates: IntCounter,
     /// number of items in cache that are outdated
     native_price_cache_outdated_entries: IntGauge,
+    /// Tracks the CoinGecko batch size
+    coin_gecko_batch_size: IntCounter,
```

Comment:
These metrics are only relevant to coingecko so they should live there IMO. That way you also don't have to increase the visibility of his `Metrics` struct.